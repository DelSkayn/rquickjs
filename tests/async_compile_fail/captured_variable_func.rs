use rquickjs::{async_with, AsyncContext, AsyncRuntime};

pub async fn test() {
    let rt = AsyncRuntime::new().unwrap();
    let ctx = AsyncContext::full(&rt).await.unwrap();

    let mut var = 1u32;
    let var_ref = &mut var;
    async_with!(ctx => |ctx|{
        ctx.spawn(async move {
            *var_ref += 1;
        })
    })
    .await
}

fn main() {}
