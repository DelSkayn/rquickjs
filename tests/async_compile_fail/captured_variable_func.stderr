error[E0597]: `var` does not live long enough
  --> tests/async_compile_fail/captured_variable_func.rs:8:19
   |
7  |       let mut var = 1u32;
   |           ------- binding `var` declared here
8  |       let var_ref = &mut var;
   |                     ^^^^^^^^ borrowed value does not live long enough
9  |       async_with!(ctx => |ctx|{
10 | /         ctx.spawn(async move {
11 | |             *var_ref += 1;
12 | |         })
   | |__________- argument requires that `var` is borrowed for `'static`
...
15 |   }
   |   - `var` dropped here while still borrowed
